### PR TITLE
Use rsync to periodically copy system logs to the SD card [DEVC-446]

### PR DESCRIPTION
**Pull request by silverjam**
Recast of Dennis' PR #513 to add functionality to copy system logs periodically.  This also add support for migrating the the filesystem of the sdcard to f2fs to avoid filesystem corruption.

Still need to add ping logging support and will increase the ping frequency to 2 Hz.

The F2FS functionality is trigger when the `logging_file_system` setting is set and settings are persisted (and the device is rebooted).  The `migrate_sdcard` service will then partition any file system it finds which has (1) at least one partition and (2) the first partition is not already an F2FS partition.  After the F2FS migration is complete, the device restarts.

The `copy_sys_logs` service only starts at the moment if the logging filesystem is already F2FS.  This avoids having to deal with service lifecycle problems during the filesystem migration.

This exposes several new advanced options in the console:
+ standalone logging -> logging file system
+ standalone logging -> copy system logs

And for the ping logging (and frequency selection) support:
+ system -> connectivity check frequency - how often ping commands are issued
+ system -> connectivity retry frequency - how often we retry ping in-between failures
+ system -> log ping activity - if we should log the output of the ping command

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018041322-8-gd96909c
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=498893)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-gd96909c&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-gd96909c&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018041322-8-gd96909c
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018041322-8-gd96909c)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018041322-8-gd96909c/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018041322-8-gd96909c

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018041322-10-gaf374f8
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=499505)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-10-gaf374f8&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-10-gaf374f8&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018041322-10-gaf374f8
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018041322-10-gaf374f8)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018041322-10-gaf374f8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018041322-10-gaf374f8

File: package/piksi_system_daemon/piksi_system_daemon/src/main.c [hunk 1]
```diff
@@ -129,6 +129,16 @@ static int uart_configure(const uart_t *uart)
   return 0;
 }
 
+static int logging_filesystem_notify(void* context)
```
Comment (benjaminaltieri) [inline package/piksi_system_daemon/piksi_system_daemon/src/main.c]: Looks like this isn't being used.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018041322-13-gd86c109
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=499589)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-13-gd86c109&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-13-gd86c109&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018041322-13-gd86c109
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018041322-13-gd86c109)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018041322-13-gd86c109/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018041322-13-gd86c109

File: board/piksiv3/rootfs-overlay/etc/init.d/copy_sys_logs.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+source /etc/init.d/logging.sh
+
+should_run()
+{
+  if needs_migration $MOUNTNAME; then
+    logw --sbp "Exiting: the SD card needs to be migrated..."
+    return 1
+  fi
+
+  if [[ -n "${COPY_SYS_LOGS}" ]]; then
```
Comment (benjaminaltieri) [inline board/piksiv3/rootfs-overlay/etc/init.d/copy_sys_logs.sh]: Testing this and this check does not seem to do what it aims. If "copy_system_logs=True" is not in my config.ini and I set "copy_system_logs" to True dynamically, the script still exits with "Exiting: logging is not enabled..."

**Comment by silverjam** [comment]
Note: rebased to clean-up the commit log, no (material) changes to the code.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018041322-8-g355634d
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=499830)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-g355634d&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-g355634d&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018041322-8-g355634d
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018041322-8-g355634d)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018041322-8-g355634d/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018041322-8-g355634d

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018041322-8-g355634d
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=499883)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-g355634d&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018041322-8-g355634d&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018041322-8-g355634d
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018041322-8-g355634d)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018041322-8-g355634d/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018041322-8-g355634d

**Pull request merged**